go/tools/releaser/prepare.go

```diff
@@ -0,0 +1,255 @@
+	release, err := gh.getReleaseByTagIncludingDraft(ctx, "bazelbuild", "rules_go", version)
+	if err != nil && errors.Is(err, errReleaseNotFound) {
+		return err
+	}
+	if release != nil && !release.GetDraft() {
+		return fmt.Errorf("release %s was already published", version)
+	}
+
+	// Check that RULES_GO_VERSION is set correctly on the release branch.
+	// If this is a minor release (x.y.0), create the release branch if it
+	// does not exist.
+	fmt.Fprintf(stderr, "checking RULES_GO_VERSION...\n")
+	rootDir, err := repoRoot()
+	if err != nil {
+		return err
+	}
+	if err := checkNoGitChanges(ctx, rootDir); err != nil {
+		return err
+	}
+	majorMinor := semver.MajorMinor(version)
+	isMinorRelease := semver.Canonical(majorMinor) == version
+	branchName := "release-" + majorMinor[len("v"):]
+	if !gitBranchExists(ctx, rootDir, branchName) {
+		if !isMinorRelease {
+			return fmt.Errorf("release branch %q does not exist locally. Fetch it, set RULES_GO_VERSION, add commits, and run this command again.")
+		}
+		if err := checkRulesGoVersion(ctx, rootDir, "HEAD", version); err != nil {
+			return err
+		}
+		fmt.Fprintf(stderr, "creating branch %s...\n", branchName)
+		if err := gitCreateBranch(ctx, rootDir, branchName, "HEAD"); err != nil {
+			return err
+		}
+	} else {
+		if err := checkRulesGoVersion(ctx, rootDir, branchName, version); err != nil {
+			return err
+		}
+	}
+
+	// Create an archive.
+	fmt.Fprintf(stderr, "creating archive...\n")
+	arcFile, err := os.CreateTemp("", "rules_go-%s-*.zip")
+	if err != nil {
+		return err
+	}
+	arcName := arcFile.Name()
+	arcFile.Close()
+	defer func() {
+		if rerr := os.Remove(arcName); err == nil && rerr != nil {
+			err = rerr
+		}
+	}()
+	if err := gitCreateArchive(ctx, rootDir, branchName, arcName); err != nil {
+		return err
+	}
+	arcSum, err := sha256SumFile(arcName)
+	if err != nil {
+		return err
+	}
+
+	// Read release notes, append boilerplate.
+	rnotesData, err := os.ReadFile(rnotesPath)
+	if err != nil {
+		return err
+	}
+	rnotesData = bytes.TrimSpace(rnotesData)
+	goVersion, err := findLatestGoVersion()
+	if err != nil {
+		return err
+	}
+	boilerplate := genBoilerplate(version, arcSum, goVersion)
+	rnotesStr := string(rnotesData) + "\n\n## `WORKSPACE` code\n\n```\n" + boilerplate + "\n```\n"
+
+	// Push the release branch.
+	fmt.Fprintf(stderr, "pushing branch %s to origin...\n", branchName)
+	if err := gitPushBranch(ctx, rootDir, branchName); err != nil {
+		return err
+	}
+
+	// Upload to mirror.bazel.build.
+	arcGHURLWithoutScheme := fmt.Sprintf("github.com/bazelbuild/rules_go/releases/download/%[1]s/rules_go-%[1]s.zip", version)
+	if uploadToMirror {
+		fmt.Fprintf(stderr, "uploading archive to mirror.bazel.build...\n")
+		if err := copyFileToMirror(ctx, arcGHURLWithoutScheme, arcName); err != nil {
+			return err
+		}
+	}
+
+	// Create or update the GitHub release.
+	if release == nil {
+		fmt.Fprintf(stderr, "creating draft release...\n")
+		prerelease := true
```

Comment:
Why prerelease?

Related issue:
https://github.com/bazelbuild/rules_go/issues/2899

Reply:
This tool will no longer set the prerelease bit. I don't think there was any particular reason for it, but from #2899, it sounds like it was causing issues. The version number should express whether something is prerelease or not.